### PR TITLE
Improve types and tweak some names for refresh tokens

### PR DESCRIPTION
**Pull request by bennostein**

### CODE REVIEW

File: skipruntime-ts/ts/src/skipruntime_service.ts [hunk 1]
```diff
@@ -51,13 +52,14 @@ export type OutputDefinition = {
...
 
 export interface GenericSkipService {
-  // name / duration in milliseconds
-  tokens?: Record<string, number>;
   localInputs(): Record<string, InputDefinition>;
   remoteInputs(): Record<string, RemoteInputs>;
 
   outputs(): Record<string, OutputDefinition>;
 
+  // name / duration in milliseconds
```
Comment (jberdine) [inline skipruntime-ts/ts/src/skipruntime_service.ts:60]: Both name and duration are present, right, not one or the other? If so:
```suggestion
  // name, duration in milliseconds
```

File: skipruntime-ts/ts/src/skipruntime_service.ts [hunk 2]
```diff
@@ -66,14 +68,15 @@ export interface GenericSkipService {
...
   database: Database;
-  inputs: string[];
+  tables: string[];
 };
 
 export interface SimpleSkipService {
   inputTables?: string[];
   remoteTables?: Record<string, SimpleRemoteInputs>;
+
   // name / duration in milliseconds
```
Comment (jberdine) [inline skipruntime-ts/ts/src/skipruntime_service.ts:78]: same / vs , here

File: skipruntime-ts/ts/src/skipruntime_api.ts [hunk 3]
```diff
@@ -16,6 +16,8 @@ export type TTableCollection = any;
 export type TTable = any;
 export type Param = any;
 
+export type RefreshToken = number & { __type_brand: "SkipRefreshToken" };
```
Comment (jberdine) [inline skipruntime-ts/ts/src/skipruntime_api.ts]: The implementation of `freeze` does `Object.defineProperty(value, "__sk_frozen"` which seems to be playing a similar role. Maybe it is worthwhile to use just one mechanism in both places?

File: skipruntime-ts/ts/src/skipruntime_runner.ts [hunk 4]
```diff
@@ -95,10 +96,10 @@ function toWriters(
 }
 
 class SimpleToGenericSkipService implements GenericSkipService {
-  tokens?: Record<string, number>;
+  refreshTokens?: Record<string, RefreshToken>;
```
Comment (jberdine) [inline skipruntime-ts/ts/src/skipruntime_runner.ts:99]: I don't understand the distinction between `Token`s and `RefreshToken`s. Each Token has a name and duration, and an array of them seems isomorphic to this `Record<string, RefreshToken>` which is like an association list right?

File: skipruntime-ts/ts/src/skipruntime_api.ts (hunk 3 above)
Comment (mbouaziz) [reply skipruntime-ts/ts/src/skipruntime_api.ts]: No, here, it's a hack for opaque types, it doesn't convey any value.

Feel free to use `Internal.Opaque<number, "SkipRefreshToken">`

Comment (bennostein) [reply skipruntime-ts/ts/src/skipruntime_api.ts]: Right, the idea here is not to enforce immutability but to make RefreshToken an opaque type that one can't accidentally mix up with `number`.  I followed what seems like TS ecosystem standard of "branded" types but will switch to using Mehdi's Opaque type constructor instead if that achieves the same!

File: skipruntime-ts/ts/src/skipruntime_runner.ts (hunk 4 above)
Comment (bennostein) [reply skipruntime-ts/ts/src/skipruntime_runner.ts:99]: `Token` here is an implementation detail of the `TimedQueue` which manages timing the token refreshes; users should only be dealing with `RefreshToken`.  I've now renamed and shuffled things around to make that more clear and avoid the name clash.

The types are indeed isomorphic but they have different interpretations: a `Token` (now `TQ_Token`) is an identifier paired with a _duration_ which the `TimedQueue` uses to schedule periodic refreshes. A `RefreshToken` on the other hand is a handle on the actual value being refreshed -- set to the current timestamp on refresh, though it should just be used as an opaque dependency in the computation graph, e.g. to trigger polls of external non-reactive dependencies.

**Pull request merged**